### PR TITLE
Update kube-version on harvester-csi-driver

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
Issue: https://github.com/rancher/rancher/issues/41515

### CODE REVIEW

**Comment by Vicente-Cheng** [comment]
~Hi @geethub97, 
I saw the CI error comes as follows:~
```
ERRO[1785] The following new assets have been introduced: map[harvester-csi-driver:[100.0.4+up0.1.16]]
```

~I trace this version and found that it is the forward port from v2.6.
How should I handle this case on v2.7 branch?~

~I thought for v2.7. We will use v102.0.x version of harvester csi driver?
Thanks!~

Update: I already fixed it. Missing some changes on the current `release.yaml`. Thanks!

File: packages/harvester/harvester-csi-driver/generated-changes/patch/Chart.yaml.patch [hunk 1]
```diff
@@ -1,6 +1,11 @@
 --- charts-original/Chart.yaml
 +++ charts/Chart.yaml
-@@ -5,9 +5,10 @@
+@@ -1,13 +1,14 @@
+ annotations:
+   catalog.cattle.io/certified: rancher
+   catalog.cattle.io/display-name: Harvester CSI Driver
+-  catalog.cattle.io/kube-version: '>= 1.18.0-0 < 1.26.0-0'
++  catalog.cattle.io/kube-version: '>= 1.18.0-0 < 1.27.0-0'
```
Comment (yaocw2020) [inline packages/harvester/harvester-csi-driver/generated-changes/patch/Chart.yaml.patch]: catalog.cattle.io/kube-version: '>= 1.23.0-0 < 1.27.0-0'

Comment (Vicente-Cheng) [reply packages/harvester/harvester-csi-driver/generated-changes/patch/Chart.yaml.patch]: Nice catch! Thanks!

Comment (Vicente-Cheng) [reply packages/harvester/harvester-csi-driver/generated-changes/patch/Chart.yaml.patch]: fixed!

**Review by diogoasouza — APPROVED** [review]

**Pull request merged**